api/v1alpha1/crds/caren.nutanix.com_awsclusterconfigs.yaml

```diff
@@ -298,6 +298,23 @@ spec:
                         type: string
                     type: object
                 type: object
+              encryption:
+                description: |-
+                  Encryption defines the configuration to enable encryption at REST
+                  This configuration is used by API server to encrypt data before storing it in ETCD.
+                  Currently the encryption only enabled for secrets and configmaps.
+                properties:
+                  providers:
+                    default: aescbc
+                    description: Encryption providers
+                    enum:
+                    - aescbc
+                    - aesgcm
+                    items:
+                      type: string
```

Comment:
Don't these providers require keys to be provided too?

Reply:
in konvoy we generate the keys and store in secrets. Should we do the same here?

Reply:
@supershal are you suggesting that clients(konvoy, kommander UI) need to create these Secrets?

Reply:
@dkoshkin I am proposing that CAREN creates the Encryption key secrets and Create a secret that contains `EncryptionConfig` . Similar as how konvoy does it.

Reply:
👍 SGTM, thanks for clarifying @supershal

Reply:
> Ok let's do the same for first cut, but allow for potential extension later by making this an object with a single name field? Or a map from string to object (with no properties for now).

Let me try this out.

Reply:
I think this should be explicit in the API and it should be possible to specify an existing secret or secret name. We can always default to generate as long as it can be disabled.

Reply:
Both `aesgcm` and `aescbc` requires a secret key or a set of secret keys in terms of rotations. The KMS v2 would require configuration to support KMS plugin's socket location and timeouts and cache size.

A one time secret key (a unique slice of bytes) for `aesgcm` and `aescbc` can be generated from CAREN.
So far we do not allow passing secret in the CAREN API. Providing K8s Secret object's reference would make more sense if we decide to go that route. Then we have to come up with secret format and lifecycle of the secret, specially when we introduce rotating of the keys. 

Given the fact that
1) `aescbc` and `aesgcm` are not recommended because former is weak and latter requires rotation implemented as it must rotate key after 200k writes.
2) we have not implemented or planning to implement rotation of the keys. If we plan to implement rotation at some point, getting secret keys from user could become complex.
3) The recommended provider KMS V2 provides key rotation

Should we provide a way for user to create a secret and provide its reference in CAREN API?
In future, what KMS V2 configuration we would allow user to modify. (cache, socket path, timeouts) Or CAREN will handle it using some sane defaults in its internal configuration.

@jimmidyson we briefly discussed some of it in scrum. I will setup a huddle tomorrow so that we can finalize it.